tests/robustness/model/deterministic.go

```diff
@@ -146,21 +146,28 @@ func (s etcdState) step(request EtcdRequest) (etcdState, EtcdResponse) {
 					KVs: []KeyValue{},
 				}
 				if op.WithPrefix {
+					var count int64
 					for k, v := range s.KeyValues {
 						if strings.HasPrefix(k, op.Key) {
 							opResp[i].KVs = append(opResp[i].KVs, KeyValue{Key: k, ValueRevision: v})
+							count += 1
 						}
 					}
 					sort.Slice(opResp[i].KVs, func(j, k int) bool {
 						return opResp[i].KVs[j].Key < opResp[i].KVs[k].Key
 					})
+					if op.Limit != 0 && count > op.Limit {
+						opResp[i].KVs = opResp[i].KVs[:op.Limit]
```

Comment:
Should we update the `count` in such case? 
```suggestion
						opResp[i].KVs = opResp[i].KVs[:op.Limit]
						count = op.Limit
```

Reply:
No, count is the total count of entries, it intentionally can be higher then response size to inform client that they should paginate.